machinery/gateway_api_topology.go

```diff
@@ -170,6 +256,110 @@ func NewGatewayAPITopology(options ...GatewayAPITopologyOptionsFunc) *Topology {
 		}
 	}
 
+	if o.ExpandGRPCRouteRules {
+		grpcRouteRules := lo.FlatMap(o.GRPCRoutes, GRPCRouteRulesFromGRPCRouteRule)
+		opts = append(opts, WithTargetables(grpcRouteRules...))
+		opts = append(opts, WithLinks(LinkGRPCRouteToGRPCRouteRuleFunc())) // GRPCRoute -> GRPCRouteRule
+
+		if o.ExpandServicePorts {
+			servicePorts := lo.FlatMap(o.Services, ServicePortsFromBackendFunc)
+			opts = append(opts, WithTargetables(servicePorts...))
+			opts = append(opts, WithLinks(
+				LinkGRPCRouteRuleToServicePortFunc(grpcRouteRules),   // GRPCRouteRule -> ServicePort
+				LinkGRPCRouteRuleToServiceFunc(grpcRouteRules, true), // GRPCRouteRule -> Service
+			))
+		} else {
+			opts = append(opts, WithLinks(LinkGRPCRouteRuleToServiceFunc(grpcRouteRules, false))) // GRPCRouteRule -> Service
+		}
+	} else {
+		if o.ExpandServicePorts {
+			opts = append(opts, WithLinks(
+				LinkGRPCRouteToServicePortFunc(o.GRPCRoutes),   // GRPCRoute -> ServicePort
+				LinkGRPCRouteToServiceFunc(o.GRPCRoutes, true), // GRPCRoute -> Service
+			))
+		} else {
+			opts = append(opts, WithLinks(LinkGRPCRouteToServiceFunc(o.GRPCRoutes, false))) // GRPCRoute -> Service
+		}
+	}
+
+	if o.ExpandTCPRouteRules {
+		tcpRouteRules := lo.FlatMap(o.TCPRoutes, TCPRouteRulesFromTCPRouteFunc)
+		opts = append(opts, WithTargetables(tcpRouteRules...))
+		opts = append(opts, WithLinks(LinkTCPRouteToTCPRouteRuleFunc())) // TCPRoute - TCPRouteRules
+
+		if o.ExpandServicePorts {
+			servicePorts := lo.FlatMap(o.Services, ServicePortsFromBackendFunc)
```

Comment:
Do you mind please renaming this mapping func? In a previous PR I renamed "Backend" to "Service", but missed this one.

```suggestion
			servicePorts := lo.FlatMap(o.Services, ServicePortsFromServiceFunc)
```

Reply:
Sure, I'll rename this 👍